src/cascadia/TerminalApp/Utils.h

```diff
@@ -30,3 +30,69 @@ inline std::string JsonKey(const std::string_view key)
 }
 
 winrt::Windows::UI::Xaml::Controls::IconElement GetColoredIcon(const winrt::hstring& path);
+
+// Helper to establish an ordering on guids. This does _NOT_ work to check if
+// two guids are the same. Use this with a std::set for determining if GUIDS are
+// unique
+struct GuidOrdering
+{
+    bool operator()(const GUID& lhs, const GUID& rhs) const
+    {
+        return memcmp(&lhs, &rhs, sizeof(rhs)) < 0;
+    }
+};
+
+// Helper to establish if two GUIDs are equal.
+struct GuidEquality
+{
+    bool operator()(const GUID& lhs, const GUID& rhs) const
+    {
+        return memcmp(&lhs, &rhs, sizeof(rhs)) == 0;
+    }
+};
+
+namespace TerminalApp
+{
+    class JsonUtils;
+};
+
+class TerminalApp::JsonUtils final
+{
+public:
+    static void GetOptionalColor(const Json::Value& json,
+                                 std::string_view key,
+                                 std::optional<uint32_t>& color);
+
+    static void GetOptionalString(const Json::Value& json,
+                                  std::string_view key,
+                                  std::optional<std::wstring>& target);
+
+    static void GetOptionalGuid(const Json::Value& json,
+                                std::string_view key,
+                                std::optional<GUID>& target);
+
+    static void GetOptionalDouble(const Json::Value& json,
+                                  std::string_view key,
+                                  std::optional<double>& target);
+
+    // Be careful, if you pass a lambda stright into this, it will explode in

```

Comment:
this is almost certainly not right; wasn't it because we were doing the specialization elsewhere?